base/linalg/bunchkaufman.jl

```diff
@@ -22,36 +22,38 @@ BunchKaufman(A::AbstractMatrix{T}, ipiv::Vector{BlasInt}, uplo::Char, symmetric:
 `bkfact!` is the same as [`bkfact`](@ref), but saves space by overwriting the
 input `A`, instead of creating a copy.
 """
-function bkfact!(A::StridedMatrix{<:BlasReal}, uplo::Symbol = :U,
-    symmetric::Bool = issymmetric(A), rook::Bool = false)
-
-    if !symmetric
-        throw(ArgumentError("Bunch-Kaufman decomposition is only valid for symmetric matrices"))
+function bkfact!(A::HermOrSym{T,S} where {T<:BlasReal,S<:StridedMatrix{T}}, rook::Bool = false)
+    if rook
+        LD, ipiv, info = LAPACK.sytrf_rook!(A.uplo, A.data)
+    else
+        LD, ipiv, info = LAPACK.sytrf!(A.uplo, A.data)
     end
+    BunchKaufman(LD, ipiv, A.uplo, true, rook, info)
+end
+function bkfact!(A::Symmetric{T,S} where {T<:BlasComplex,S<:StridedMatrix{T}}, rook::Bool = false)
     if rook
-        LD, ipiv, info = LAPACK.sytrf_rook!(char_uplo(uplo), A)
+        LD, ipiv, info = LAPACK.sytrf_rook!(A.uplo, A.data)
     else
-        LD, ipiv, info = LAPACK.sytrf!(char_uplo(uplo), A)
+        LD, ipiv, info = LAPACK.sytrf!(A.uplo, A.data)
     end
-    BunchKaufman(LD, ipiv, char_uplo(uplo), symmetric, rook, info)
+    BunchKaufman(LD, ipiv, A.uplo, true, rook, info)
 end
```

Comment:
If I read the diff correctly (quite possibly not!), the two preceding `bkfact!` method bodies are identical? If so, perhaps worth DRYing out?